### PR TITLE
feat: one-time classification backfill via POST /classify-pending

### LINKED ISSUE #145: One-time classification backfill for existing entries (status + kind)
## Problem

Issues #119 (status: canonical/draft/deprecated) and #12 (episodic/semantic `kind`) classify memories **at capture time** via the `classifyEntry` LLM call. Existing entries created before those features shipped stay unclassified (no `status:` or `kind:` tag).

This is intentionally **not** a correctness problem — unclassified entries behave exactly as today (see #119 plan). It's an optional quality improvement for installs with a large existing history that want the back-catalog typed and protected.

## Proposed Solution

A **one-time, opt-in classification backfill** that runs `classifyEntry` over existing entries and writes both `status:canonical` (where warranted) and `kind:episodic|semantic` — in a single pass, so history is only walked once.

### Important: this is NOT a schema migration
No DDL. It only writes tags. The real cost/risk is **one LLM call per existing entry** — on a large store this hits Workers AI rate limits, Worker CPU-time limits, and token cost.

### Implementation notes
- Mirror the existing `/vectorize-pending` pattern (`src/index.ts:1855`): a `POST /classify-pending` endpoint that processes a **bounded batch per call** (e.g. ≤25 entries), is **idempotent** (skips entries that already have a `status:`/`kind:` tag) and **resumable** (safe to stop/restart).
- A thin driver (npm script or cron step) calls it until drained.
- Conservative canonical promotion, identical to the capture-time classifier (only clearly settled decisions / durable facts / permanent preferences).
- Reuse `classifyEntry`, `getStatus`, `withStatus` from #119 and the `kind` helpers from #12 — only promote where the current status is unset.

### Priority / dependencies
- **Depends on #119 and #12** (needs `classifyEntry` returning `{importance, canonical, kind}`).
- **Low priority, off the critical path.** Nobody needs this for #119 or #12 to be valuable; it's a nice-to-have for existing large histories.

### PR DESCRIPTION
**Pull request by Aneesh-382005**
Closes #145.

This adds a one-time, opt-in backfill for entries that were captured before the status (#119) and kind (#12) classification features existed, so they never got a status: or kind: tag.

POST /classify-pending picks up to 25 untagged entries, runs the existing classifyEntry on each, and writes back whatever tags it resolves (kind:episodic/kind:semantic, and status:canonical only if the entry doesn't already have a status). It only ever touches tags without making any schema changes or modifying any other fields

It mirrors /vectorize-pending's shape exactly: bounded batch per call, skips anything already tagged (idempotent), safe to call repeatedly until remaining hits 0 (resumable).

**And yeah:** 
if an entry's classification is genuinely ambiguous (no kind, not canonical), it won't get tagged and will show up as "remaining" again next call. That's intentional per the issue's own definition of idempotency (tag presence, not a separate "attempted" marker), expected.

Added test/integration/classify-pending.test.ts (9 tests: auth, empty state, tagging + full drain, skip-already-tagged, resumability, the ambiguous case above, status not clobbered, importance_score untouched, partial-batch failure). tsc --noEmit and the full suite pass (435/435).

* **Note:**
This PR was written with help from an AI agent (I reviewed it). Happy to walk through any of it. First time contributing here, I'd appreciate any feedback. If anything's off about the approach, the tests, or how it fits with the rest of the codebase, I'd love to hear it and redo.

### CODE REVIEW

**Comment by rahilp** [comment]
Hey @Aneesh-382005, thank you for this. Genuinely nice first contribution. The idempotent, resumable, bounded-batch design mirroring `/vectorize-pending` is exactly the right pattern, and the tests are thorough. I especially liked that you documented the ambiguous "stays remaining" case out in the open instead of hiding it. I pulled it down and ran the suite locally and everything checks out.

One thing I'd love your thoughts on before it fully lands. Right now `/classify-pending` is only reachable by manually POSTing to it, so it's effectively a hidden feature. The people who most need the backfill (anyone who deployed before the `kind:`/`status:` tags existed) would never know to call it.

The nicest fix would be to surface it in the dashboard the same way vectorize already does. There's a "Not indexed" section in `public/index.html` (see `renderVectorizeSection` and the button that loops `POST /vectorize-pending`) that shows a count and drains the backfill straight from the UI. A parallel section that shows how many entries still lack `kind:`/`status:` tags and runs `/classify-pending` the same way would make this discoverable and self-serve.

Would you be interested in taking a crack at that as part of this PR? Absolutely no pressure, and no worries at all if you'd rather keep this one focused on the endpoint and leave the UI as a follow-up. Either way I really appreciate the work you put in here.

Thanks again!

**Comment by Aneesh-382005** [comment]
Hey @rahilp, thank you for taking the time to review it.

The dashboard section has been added, as you suggested. It mirrors `renderVectorizeSection`/`runVectorize`: `/stats` now returns a count of `unclassified` entries. The dashboard shows a "Not classified" card with a "Classify now" button whenever there are unclassified entries. Clicking it loops `/classify-pending` until fully drained, then refreshes stats so the card disappears.

Added two tests for the new `/stats` field.

Let me know if you'd like anything adjusted : )

File: public/index.html [hunk 1]
```diff
@@ -3258,6 +3260,52 @@ <h3>Second Brain</h3>
...
+          while (remaining > 0) {
+            const res = await fetch(`${WORKER_URL}/classify-pending`, {
+              method: 'POST',
+              headers: { Authorization: `Bearer ${AUTH_TOKEN}` }
+            })
+            if (!res.ok) throw new Error(`Server error: ${res.status}`)
+            const data = await res.json()
+            remaining = data.remaining ?? 0
+            totalProcessed += data.processed ?? 0
+            if ((data.processed ?? 0) === 0 && remaining > 0) break
```
Comment (rahilp) [inline public/index.html]: This loop can spin forever in one case. /classify-pending counts an entry as processed even when classifyEntry returns no kind or status, so ambiguous entries stay in remaining while processed stays above 0. The processed === 0 guard never trips, so it keeps calling the endpoint (and the AI) every iteration. Two ways it shows up: the tail of a backfill where the only entries left are ones the model won't classify, and a Workers AI outage where nothing drains at all.

Could we break on lack of forward progress instead? Something like:

```
let prevRemaining = Infinity
while (remaining > 0) {
  // ...fetch + parse...
  remaining = data.remaining ?? 0
  totalProcessed += data.processed ?? 0
  if (remaining >= prevRemaining) break   // no progress this round, stop
  prevRemaining = remaining
}
```

That drains everything it can and stops cleanly when a round makes no progress. Really nice work on the rest of it, the dashboard section came out great.

**Review by rahilp — CHANGES_REQUESTED** [review]

File: public/index.html (hunk 1 above)
Comment (Aneesh-382005) [reply public/index.html]: You're right, I totally missed this.
I believe it's fixed now.
Swapped in your non-decreasing-remaining check, and it stops after 2 calls instead of looping.

**Review by rahilp — APPROVED** [review]

**Comment by rahilp** [comment]
@Aneesh-382005 this turned out great, thank you. You took it from a solid endpoint to a genuinely polished feature: the dashboard card, the `/stats` count, and the progress-guarded loop all came together cleanly, and you turned each round of feedback around fast and thoughtfully. The termination fix in particular was exactly right.

I really appreciate the care you put in here, especially the thorough tests and the way you documented the edge cases in the open instead of papering over them. This is a strong first contribution and it is landing in the app. Hope to see more from you, and thanks again for making Second Brain better.

**Comment by Aneesh-382005** [comment]
Thank you, @rahilp . This was a learning experience and I appreciate you taking the time. Happy to contribute and I look forward to collaborating more.
